pkg/encryption/path_new.go

```diff
@@ -0,0 +1,300 @@
+// Copyright (C) 2019 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package encryption
+
+import (
+	"crypto/hmac"
+	"crypto/sha512"
+	"encoding/base64"
+	"strings"
+
+	"github.com/zeebo/errs"
+
+	"storj.io/storj/pkg/paths"
+	"storj.io/storj/pkg/storj"
+)
+
+// StoreEncryptPath encrypts the path using the provided cipher and looking up
+// keys from the provided store and bucket.
+func StoreEncryptPath(bucket string, path paths.Unencrypted, cipher storj.Cipher, store *Store) (
```

Comment:
This function name is quite confusing. Initially I though it would store and encypted path somewhere. `EncryptPathWithStore` or similar would be better.

Reply:
I agree, that would be a better name. I hope to complete the refactor and have only one set of encryption methods, and it can then use the old name.